### PR TITLE
Apply example_ynh

### PR DESCRIPTION
**Pull request by Tagadda**
## Problem

- I started to use this app :eyes: 

## Solution

- Apply last example_ynh

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by Tagadda** [comment]
!testme

**Comment by Tagadda** [comment]
!testme

**Comment by Tagadda** [comment]
!testme

**Comment by Tagadda** [comment]
The config panel will come in a future PR.

File: scripts/change_url [hunk 1]
```diff
@@ -64,7 +65,7 @@ fi
 #=================================================
 # MODIFY URL IN NGINX CONF
 #=================================================
-ynh_script_progression --message="Updating NGINX web server configuration..." --weight=3
+ynh_script_progression --message="Updating NGINX web server configuration..." --weight=1
```
Comment (Tagadda) [inline scripts/change_url]:
```suggestion
ynh_script_progression --message="Updating NGINX web server configuration..." --weight=3
```

File: scripts/install [hunk 2]
```diff
@@ -55,13 +87,12 @@ ynh_system_user_create --username=$app --home_dir="$final_path"
 #=================================================
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
-ynh_script_progression --message="Setting up source files..." --weight=4
+ynh_script_progression --message="Setting up source files..." --weight=1
```
Comment (Tagadda) [inline scripts/install]:
```suggestion
ynh_script_progression --message="Setting up source files..." --weight=4
```

File: scripts/remove [hunk 3]
```diff
@@ -19,20 +20,55 @@ domain=$(ynh_app_setting_get --app=$app --key=domain)
...
+#=================================================
+# ...
 #=================================================
-ynh_script_progression --message="Removing $app main directory..." --weight=3
 
-# Remove sources
+#=================================================
+# REMOVE APP MAIN DIR
+#=================================================
+ynh_script_progression --message="Removing app main directory..." --weight=1
```
Comment (Tagadda) [inline scripts/remove]:
```suggestion
ynh_script_progression --message="Removing app main directory..." --weight=3
```

File: scripts/restore [hunk 4]
```diff
@@ -53,22 +60,44 @@ ynh_system_user_create --username=$app --home_dir="$final_path"
 #=================================================
 # RESTORE THE APP MAIN DIR
 #=================================================
-ynh_script_progression --message="Restoring the app main directory..." --weight=4
+ynh_script_progression --message="Restoring the app main directory..." --weight=1
```
Comment (Tagadda) [inline scripts/restore]:
```suggestion
ynh_script_progression --message="Restoring the app main directory..." --weight=4
```

File: scripts/remove [hunk 5]
```diff
@@ -19,20 +20,55 @@ domain=$(ynh_app_setting_get --app=$app --key=domain)
...
+
+# Remove the app directory securely
 ynh_secure_remove --file="$final_path"
```
Comment (Tagadda) [inline scripts/remove:34]: We should do this only if we have the `--purge` !

File: scripts/remove [hunk 6]
```diff
@@ -19,20 +20,55 @@ domain=$(ynh_app_setting_get --app=$app --key=domain)
...
+
+#=================================================
+# SPECIFIC REMOVE
+#=================================================
+# REMOVE VARIOUS FILES
+#=================================================
+ynh_script_progression --message="Removing various files..." --weight=1
+
+# Remove a directory securely
+ynh_secure_remove --file="/etc/$app"
```
Comment (Tagadda) [inline scripts/remove]: This does not exist ?

File: scripts/upgrade [hunk 7]
```diff
@@ -77,13 +90,12 @@ ynh_system_user_create --username=$app --home_dir="$final_path"
 
 if [ "$upgrade_type" == "UPGRADE_APP" ]
 then
-	ynh_script_progression --message="Upgrading source files..." --weight=4
+	ynh_script_progression --message="Upgrading source files..." --weight=1
```
Comment (Tagadda) [inline scripts/upgrade]:
```suggestion
	ynh_script_progression --message="Upgrading source files..." --weight=4
```

File: scripts/remove (hunk 6 above)
Comment (Tagadda) [inline scripts/remove]:
```suggestion
# ...
#=================================================

```

File: scripts/remove (hunk 5 above)
Comment (Tagadda) [inline scripts/remove]: We should delete `$final_path/_h5ai` and check for `--purge` to delete `$final_path`

File: scripts/install [hunk 8]
```diff
@@ -71,17 +102,49 @@ chown -R $app:www-data "$final_path"
...
 
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+ynh_script_progression --message="Configuring PHP-FPM..." --weight=1
+
+# Create a dedicated PHP-FPM config
+ynh_add_fpm_config
+
+#=================================================
```
Comment (ericgaspar) [inline scripts/install:101]: L.116 to L.137 can be removed I believe

File: scripts/upgrade [hunk 9]
```diff
@@ -96,6 +108,43 @@ ynh_script_progression --message="Upgrading NGINX web server configuration..." -
...
+ynh_install_app_dependencies $pkg_dependencies
+
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+ynh_script_progression --message="Upgrading PHP-FPM configuration..." --weight=1
+
+# Create a dedicated PHP-FPM config
+ynh_add_fpm_config
+
```
Comment (ericgaspar) [inline scripts/upgrade:96]: L.125 to L.147 can be removed

File: scripts/change_url [hunk 10]
```diff
@@ -90,8 +91,18 @@ then
 	ynh_store_file_checksum --file="/etc/nginx/conf.d/$new_domain.d/$app.conf"
 fi
 
+#=================================================
```
Comment (ericgaspar) [inline scripts/change_url:94]: L.94 to L.105 can be removed

File: scripts/install [hunk 11]
```diff
@@ -37,12 +42,39 @@ then
...
+# ...
+#=================================================
+
+#=================================================
+# INSTALL DEPENDENCIES
+#=================================================
+ynh_script_progression --message="Installing dependencies..." --weight=1
+
+### `ynh_install_app_dependencies` allows you to add any "apt" dependencies to the package.
```
Comment (ericgaspar) [inline scripts/install]: Same. we can remove the comments

File: scripts/remove [hunk 12]
```diff
@@ -19,20 +20,52 @@ domain=$(ynh_app_setting_get --app=$app --key=domain)
...
+
+#=================================================
+# REMOVE DEPENDENCIES
+#=================================================
+ynh_script_progression --message="Removing dependencies..." --weight=1
+
+# Remove metapackage and its dependencies
+ynh_remove_app_dependencies
+
+#=================================================
```
Comment (ericgaspar) [inline scripts/remove:60]: can be removed

**Review by ericgaspar — CHANGES_REQUESTED** [review]

File: scripts/remove (hunk 12 above)
Comment (Tagadda) [reply scripts/remove:60]: what about php ? :thinking:

File: scripts/remove (hunk 5 above)
Comment (Tagadda) [reply scripts/remove]:
```suggestion
# Remove the app directory securely
ynh_secure_remove --file="$final_path/_h5ai"
```

**Comment by Tagadda** [comment]
Hey @ericgaspar o/ I'm pretty sure this app needs PHP. Why removing it?

**Comment by ericgaspar** [comment]
You are right, we need to keep `ynh_remove_app_dependencies` as `ffmpeg` and `imagemagick` are installed but I guess we don't need `php$phpversion-fpm` to be installed... (The current version in master seems to work fine withouth installing `php-fpm.conf`)

File: scripts/install (hunk 11 above)
Comment (Tagadda) [reply scripts/install]:
```suggestion
```

**Comment by Tagadda** [comment]
!testme

File: scripts/_common.sh [hunk 13]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+#=================================================
+# COMMON VARIABLES
+#=================================================
+
+phpversion="7.3"
```
Comment (ericgaspar) [inline scripts/_common.sh]:
```suggestion
YNH_PHP_VERSION=7.4
```

File: scripts/_common.sh [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+
+# dependencies used by the app
+pkg_dependencies="php$phpversion-fpm ffmpeg imagemagick"
```
Comment (ericgaspar) [inline scripts/_common.sh]:
```suggestion
pkg_dependencies="php$YNH_PHP_VERSION-fpm ffmpeg imagemagick"
```

File: manifest.json [hunk 15]
```diff
@@ -22,7 +23,7 @@
     "requirements": {
         "yunohost": ">= 4.3.0"
     },
-    "multi_instance": false,
+    "multi_instance": true,
     "services": [
         "nginx",
         "php7.3-fpm"
```
Comment (ericgaspar) [inline manifest.json]:
```suggestion
        "php7.4-fpm"
```

**Review by ericgaspar — COMMENTED** [review]
Let's upgrade to PHP7.4

**Comment by ericgaspar** [comment]
Do we need to increase `upload_max_file_size` and `post_max_size` ?
```
; php_admin_value[upload_max_filesize] = 50M
; php_admin_value[post_max_size] = 50M
```

**Comment by Tagadda** [comment]
Now you mention it, I guess it should be a config panel option :thinking: 

EDIT: About the comments, usually I leave them along as it's simpler for me to merge charges from example_ynh. But even with this, there was some cleaning needed.

**Comment by Tagadda** [comment]
👀

**Pull request merged**